lang/cs/Org.Apache.REEF.IMRU/OnREEF/IMRUTasks/MapTaskHost.cs

```diff
@@ -109,7 +92,7 @@ internal sealed class MapTaskHost<TMapInput, TMapOutput> : ITask, IObserver<IClo
         /// <returns></returns>
         public byte[] Call(byte[] memento)
         {
-            while (Interlocked.Read(ref _shouldCloseTask) == 0)
+            while (!_taskCloseHandler.ShouldCloseTask())
```

Comment:
Can you put the `break` condition check here instead of using the `break` keyword?